### DISCUSSION TITLE
Line endings: What's the diff between "Set" and "Convert"?

### DISCUSSION THREAD

**Discussion by humdingerb** [Genio-The-Haiku-IDE/Genio#87] [Q&A]
It appears that the "Set *" items in the "Edit|Line endings" menu don't do anything. Or am I using them incorrectly?
The "Convert *" items do work.

I wonder what the intended difference between Set and Convert could be. If there isn't any actually, I propose renaming the submenu items under "Line endings" simply "Unix", "DOS", "Mac".

**Comment by Freaxed** [comment]
According to scintilla documentation:

Set*:
SCI_SETEOLMODE sets the characters that are added into the document when the user presses the Enter key. You can set eolMode to one of SC_EOL_CRLF (0), SC_EOL_CR (1), or SC_EOL_LF (2). The SCI_GETEOLMODE message retrieves the current state.

Convert*:
SCI_CONVERTEOLS(int eolMode)
This message changes all the end of line characters in the document to match eolMode. Valid values are: SC_EOL_CRLF (0), SC_EOL_CR (1), or SC_EOL_LF (2).

**Comment by humdingerb** [comment]
Got it, thanks.
Is the "Set" really useful though? Converting the whole file and the using that format throughout the file is a needed feature. But keeping the line-ending format for the existing lines and use a different one for the added lines?

**Comment by Freaxed** [comment]
These features comes from Ideam and I  never really tested them.
What happen if I convert to DOS? Is the eol automatically set to DOS or
will remain UNIX? We should at least Set* accorting to the Convert*.. then
yes, we can avoid the explicit Set functions.

Il giorno lun 29 mag 2023 alle 17:17 humdinger ***@***.***>
ha scritto:

**Comment by humdingerb** [comment]
> What happen if I convert to DOS? Is the eol automatically set to DOS or will remain UNIX?

"Converting" converts all eol and new lines also get that eol format.
"Set" keeps the old eol format, only newly added lines get the set eol.

IMO, all "Set" commands can be removed, keeping the "Convert" functions and the change the "Line endings" menu to simply have the items "Unix", "DOS", "Mac".

**Comment by Freaxed** [comment]
>
> "Converting" converts all eol and new lines also get that eol format.
>
Ok!

>
> IMO, all "Set" commands can be removed, keeping the "Convert" functions
> and the change the "Line endings" menu to simply have the items "Unix",
> "DOS", "Mac".
>
I agree to your proposal, let’s simplify!

Thanks!

>
